### PR TITLE
SMC-728 - pull latest upstream master; add error handler in WebSocketRequest.js

### PR DESCRIPTION
**Pull request by cordellcalitz**

### CODE REVIEW

**Review by csampsonza — APPROVED** [review]
Would be nice to drop the fork.  But we need protection against EPIPE, .on('error' ... , and we have some pong logic -  fix for connections being dropped when inbound queue causes us to miss pongs before keepalive timeout

**Review by JohanvdWest — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -76,7 +99,7 @@ Version 1.0.21
 --------------
 *Released 2015-07-22*
 
-* Incremented and re-published to work around an aborted npm publish of v1.0.20.
+* Incrememnted and re-published to work around an aborted npm publish of v1.0.20.
```
Comment (JensEggers) [inline CHANGELOG.md:102]: new typo: 'Incremented'

File: lib/WebSocketRequest.js [hunk 2]
```diff
@@ -99,8 +99,10 @@ function WebSocketRequest(socket, httpRequest, serverConfig) {
     // Watch for the underlying TCP socket closing before we call accept
     this._socketIsClosing = false;
     this._socketCloseHandler = this._handleSocketCloseBeforeAccept.bind(this);
+    this._sockeErrorHandler = this._handleSocketErrorBeforeAccept.bind(this);
```
Comment (JensEggers) [inline lib/WebSocketRequest.js]: add missing 't' into _sockeErrorHandler

File: lib/WebSocketRequest.js [hunk 3]
```diff
@@ -499,6 +501,11 @@ WebSocketRequest.prototype._handleSocketCloseBeforeAccept = function() {
     this._removeSocketCloseListeners();
 };
 
+WebSocketRequest.prototype._handleSocketErrorBeforeAccept = function(err) {
```
Comment (JensEggers) [inline lib/WebSocketRequest.js:507]: why does this only apply to 'BeforeAccept'? are the error handlers overwritten later?

**Review by ErasmusJW — APPROVED** [review]

**Review by JensEggers — APPROVED** [review]
see comment on typos, but if these came in from upstream merge it's prob best to leave them alone

File: lib/WebSocketRequest.js (hunk 3 above)
Comment (cordellcalitz) [reply lib/WebSocketRequest.js:507]: Honestly I don't really understand it very well. It looks like this file only accepts or rejects the requests, then it gets handled on another level. So we add the error handler within this scope and remove it once it is accepted or rejected. I assume if an error can happen somewhere outside of this file that might cause an uncaught exception we would have to handle it there

**Review by DylanABacon — APPROVED** [review]

**Pull request merged**